xarray/core/indexing.py

```diff
@@ -1653,10 +1656,15 @@ class PandasIndexingAdapter(ExplicitlyIndexedNDArrayMixin):
 
     __slots__ = ("array", "_dtype")
 
-    def __init__(self, array: pd.Index, dtype: DTypeLike = None):
+    def __init__(
+        self, array: pd.Index, dtype: DTypeLike = None, *, fastpath: bool = False
+    ):
         from xarray.core.indexes import safe_cast_to_index
 
-        self.array = safe_cast_to_index(array)
+        if fastpath:
```

Comment:
is this one used?

Reply:
No. Thank you for your thorough review.

I know these micro-optimizations are annoying.

Reply:
I checked by adding and `import ipdb; ipdb.set_trace()` and running the whole test suite and wiating for it to stop.
```
pytest -s --stepwise -x
```